tests/ui/const-generics/adt_const_params/const_param_ty_impl_union.rs

```diff
@@ -0,0 +1,33 @@
+#![allow(incomplete_features)]
+#![feature(adt_const_params, structural_match)]
+
+union Union {
+    a: u8,
+}
+
+impl PartialEq for Union {
+    fn eq(&self, other: &Union) -> bool {
+        true
+    }
+}
+impl Eq for Union {}
+impl std::marker::StructuralEq for Union {}
+
+impl std::marker::ConstParamTy for Union {}
```

Comment:
xd this should error oh well not this PR's fault

Reply:
well, i'm glad i added a test at least 😆